pkg/cache/v2/resource.go

```diff
@@ -82,13 +82,13 @@ func MarshalResource(resource types.Resource) (types.MarshaledResource, error) {
 
 // GetResourceReferences returns the names for dependent resources (EDS cluster
 // names for CDS, RDS routes names for LDS).
-func GetResourceReferences(resources map[string]types.Resource) map[string]bool {
+func GetResourceReferences(resources map[string]types.ResourceWithTtl) map[string]bool {
```

Comment:
Oh yeah I like that, should simplify things quite a bit

Reply:
I spent some time trying to get this to work, the problem is that by adding a function here regular proto messages no longer satisfy the interface, which means that we end up having to wrap the resources in a struct regardless. I tried composing a `proto.Message` into `ResourceWithTtl` to make make them all just satisfy `types.Resource`, but it seems like the protobuf library does all kind of type conversions so just satisfying `proto.Message` isn't actually sufficient.

lmk if you have any ideas for how to get this working, otherwise the current approach seems like the simplest one (if verbose)

Reply:
I gave this a go, the problem here is that it means that a regular proto no longer passes as a `Resource` (at least from what I could tell), which means that all callers that currently pass `[]Resource{proto1, proto2}` would have to update it to `[]Resource{Resource{Message: proto1}, Resource{Message: proto2}`. Not sure if it would be worth the churn to change? wdyt

Reply:
I think you could do `[]Resource{{proto1}, {proto2}}` which I can live with. What do others think? I think it makes sense to align Resource with xDS Resource proto which has TTL field. But I'm fine either way.

Reply:
Omitting the field name works when there's only one field e.g. `type Resource struct { proto.Message }`. Once we introduce a second one for the TTL, they now need to be specified. 

I would expose the a function to get the TTL but that means having to embed the TTL within the proto, which then forces the user to use the `Resource` proto wrapper (not to be confused with the Go type) themselves to specify the TTL. This would be an option: we could provide a helper function to make it easier to create the proto wrapper and rely on it for holding the TTL configuration. LMK what you all think (cc @jyotimahapatra @jessicayuen)

Reply:
I'll note that this adds a bunch of complexity around anything that cares about the actual resource, like consistency checks, as we need to handle potentially unwrapping the wrapper type to see the real resource.

Reply:
Ok after much more playing around it seems like embedding `proto.Message` breaks marshalling, since the proto library makes use of some cute interface casts to do marshalling, which ends up failing when embedding the interface. As such I'd propose leaving things as is, unless someone has any ideas for how to work around this